examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,239 @@
+            _ => Ok(None),
+        }
+    } else {
+        Ok(None)
+    }
+}
+
+const fn handle_key(key: event::KeyEvent) -> Option<Message> {
+    match key.code {
+        KeyCode::Char('j') | KeyCode::Down => Some(Message::SelectNext),
+        KeyCode::Char('k') | KeyCode::Up => Some(Message::SelectPrevious),
+        KeyCode::Char('q') | KeyCode::Esc => Some(Message::Quit),
+        _ => None,
+    }
+}
+
+const fn handle_mouse(mouse: event::MouseEvent) -> Option<Message> {
+    match mouse.kind {
+        MouseEventKind::ScrollDown => Some(Message::SelectPrevious),
+        MouseEventKind::ScrollUp => Some(Message::SelectNext),
+        _ => None,
+    }
+}
+
+fn update(model: &mut Model, msg: Message) -> Option<Message> {
+    match msg {
+        Message::Quit => model.running_state = RunningState::Done,
+        Message::SelectNext => {
+            let current_index = model.table_state.selected().unwrap_or(0);
+            if current_index < model.table_items.len() - 1 {
+                model.table_state.select(Some(current_index + 1));
+            }
+        }
+        Message::SelectPrevious => {
+            let current_index = model.table_state.selected().unwrap_or(0);
+            if current_index > 0 {
+                model.table_state.select(Some(current_index - 1));
+            }
+        }
+    };
+    None
+}
+
+mod tui {
+    use std::io::stdout;
+    use std::panic;
+
+    use ratatui::backend::{Backend, CrosstermBackend};
+    use ratatui::crossterm::terminal::{
+        disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen,
+    };
+    use ratatui::crossterm::ExecutableCommand;
+    use ratatui::Terminal;
+
+    pub fn init_terminal() -> color_eyre::Result<Terminal<impl Backend>> {
+        enable_raw_mode()?;
+        stdout().execute(EnterAlternateScreen)?;
+        let terminal = Terminal::new(CrosstermBackend::new(stdout()))?;
+        Ok(terminal)
+    }
+
+    pub fn restore_terminal() -> color_eyre::Result<()> {
+        stdout().execute(LeaveAlternateScreen)?;
+        disable_raw_mode()?;
+        Ok(())
+    }
+
+    pub fn install_panic_hook() {
+        let original_hook = panic::take_hook();
+        panic::set_hook(Box::new(move |panic_info| {
+            stdout().execute(LeaveAlternateScreen).unwrap();
+            disable_raw_mode().unwrap();
+            original_hook(panic_info);
+        }));
+    }
+}
+
+fn generate_some_fans() -> Vec<Data> {
+    use fakeit::{address, contact, name};
+
+    (0..50)
+        .map(|_| {
+            let name = name::full();
+            let address = format!(
+                "{}\n{}, {} {}",
+                address::street(),
+                address::city(),
+                address::state(),
+                address::zip()
+            );
+            let email = contact::email();
+
+            Data {
+                name,
+                address,
+                email,
+            }
+        })
+        .collect()
+}
```

Comment:
Consider generating a model that has all the fields and then doing the to string parts at render time. E.g. `impl From<Person> for Row` and 
```
struct Person {
    name: String,
    address: Address,
    email: String,
}


impl Person {
    fn fake() -> Self {
        Self {
            name: fakeit::name::full(),
            address: Address::fake(),
            email: fakeit::contact::email(),
        }
    }
}

impl Address {
    fn fake() -> Self { ... }
}
```

Reply:
That may be better, but for the sake of the example, this may be easier to understand.

```rust
fn generate_some_people() -> Vec<Person> {
    (0..50)
        .map(|_| Person {
            name: name::full(),
            address: generate_fake_address(),
            email: contact::email(),
        })
        .collect()
}

fn generate_fake_address() -> String {
    format!(
        "{}\n{}, {} {}",
        address::street(),
        address::city(),
        address::state(),
        address::zip()
    )
}
```